src/wp-includes/class-wp-query.php

```diff
@@ -3099,8 +3099,8 @@ public function get_posts() {
 				$last_changed .= wp_cache_get_last_changed( 'terms' );
 			}
 
-			$cache_key = "wp_query:$key:$last_changed";
-
+			$cache_key      = "wp_query:$key:$last_changed";
+			$cached_results = false;
 			if ( null === $this->posts ) {
 				$cached_results = wp_cache_get( $cache_key, 'posts' );
```

Comment:
```suggestion
				$cached_results = wp_cache_get( $cache_key, 'posts', false, $cache_found );
```

[`$cache_found` is passed by reference](http://developer.wordpress.org/reference/functions/wp_cache_get/) according to the docs.

Reply:
+1

Reply:
+1. Pushed fix.